### PR TITLE
LT-887-third-parties

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: ui_automation_tests/features/manage_cases.feature [hunk 1]
```diff
@@ -74,14 +74,19 @@ Feature: I want to record the final decision overall on an application case
...
-  @LT_1042_can_see_ultimate_end_users
-  Scenario: Gov user can see ultimate end users in the destinations section of the case
+    
+  @LT_957_error
+  Scenario: Record decision validation
     Given I create application or application has been previously created
-    And I sign in to SSO or am signed into SSO
-    When I go to application previously created
-    # TODO improve this to test the whole case page.
-    Then I see an ultimate end user
```
Comment (phlptt) [inline ui_automation_tests/features/manage_cases.feature]: whys this been deleted

Comment (phlptt) [reply ui_automation_tests/features/manage_cases.feature]: oh i know why

File: ui_automation_tests/features/case_view_details.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  So that I can make review the case before making any decisions
+
+@LT_1042_can_see_all_parties
+  Scenario: Gov user can see all parties on the case
+    Given I create application or application has been previously created
+    And I sign in to SSO or am signed into SSO
+    When I go to application previously created
+    Then I see an ultimate end user
+    And I see a third party
+    And I see a consignee
```
Comment (phlptt) [inline ui_automation_tests/features/case_view_details.feature]: new line

File: ui_automation_tests/features/manage_cases.feature [hunk 3]
```diff
@@ -74,14 +74,19 @@ Feature: I want to record the final decision overall on an application case
...
+    And I go to internal homepage
+    When I give myself the required permissions for "Make final decisions"
+    And I go to application previously created
+    And I click record decision
+    And I "deny" application
+    And I click continue
+    And I click continue
+    Then I see error message "Select at least one denial reason"
+    And I reset the permissions
```
Comment (phlptt) [inline ui_automation_tests/features/manage_cases.feature]: this test got removed in another branch looks like you fixing the conflicts you put it back in could you remove please pal

File: ui_automation_tests/features/case_view_details.feature (hunk 2 above)
Comment (andytaylorkainos) [reply ui_automation_tests/features/case_view_details.feature]: damnit

File: ui_automation_tests/features/manage_cases.feature (hunk 3 above)
Comment (andytaylorkainos) [reply ui_automation_tests/features/manage_cases.feature]: Done

File: templates/cases/case/views/standard-destinations.html [hunk 4]
```diff
@@ -115,3 +174,64 @@ <h3 class="govuk-heading-m">Ultimate End Users</h3>
...
+									{% endif %}
+								{% endfor %}
+							</td>
+						{% endif %}
+					</tr>
+				{% endfor %}
+			</tbody>
+		</table>
+	{% endif %}
+{% endwith %}
```
Comment (Mark-Spiers-kainos) [inline templates/cases/case/views/standard-destinations.html]: newline

File: ui_automation_tests/helpers/seed_data.py [hunk 5]
```diff
@@ -254,8 +268,17 @@ def add_draft(self, draft=None, good=None, enduser=None, ultimate_end_user=None)
         data = self.request_data['ultimate_end_user'] if ultimate_end_user is None else ultimate_end_user
         ultimate_end_user_post = self.make_request('POST', url='/drafts/' + draft_id + '/ultimate-end-users/',
                                                    headers=self.export_headers, body=data)
-        ultimate_end_user_id = json.loads(ultimate_end_user_post.text)['end_user']['id']
+        ultimate_end_user_id = json.loads(ultimate_end_user_post.text)['ultimate_end_user']['id']
         self.add_ultimate_end_user_document(draft_id, ultimate_end_user_id)
+
+        data = self.request_data['consignee'] if consignee is None else consignee
```
Comment (cstobart) [inline ui_automation_tests/helpers/seed_data.py]: make variable sdicstinct

File: cases/services.py [hunk 6]
```diff
@@ -115,6 +115,21 @@ def post_case_advice(request, case_pk, json):
...
 
+    if json.get('consignee'):
+        data = base_data.copy()
+        data['consignee'] = json.get('consignee')
+        new_data.append(
+            data
+        )
+
+    if json.get('third_parties'):
+        for item in json.get('third_parties', []):
```
Comment (BenAndrew-Kainos) [inline cases/services.py]: Could this base_data.copy() and new_data section be refactored into a function?

**Pull request merged**